VERSION-RANGE-SPEC.rst

```diff
@@ -369,53 +358,44 @@ To parse a version range specifier string:
   may be strict and report an error if there are extra characters beyond "*" or
   be lenient.
 
-- Split the constraints on pipe "|". The result is a list of top-level 
-  <version-constraint> lists. Consecutive pipes should be treated as one.
+- Strip leading adn trailing pipes "|" from the constraints string.
+- Split the constraints on pipe "|". The result is a list of  <version-constraint>.
+  Consecutive pipes should be must as one and leading and trailing pipes ignored.
 
-- For each <version-constraint> list:
+- For each <version-constraint>:
 
-  - Split on comma ",". Consecutive commas  should be treated as one. The result
-    is a sub-list of <version-constraint>.
+  - Determine if the <version-constraint> starts with one of the two comparators:
 
-  - For each <version-constraint> in this sub-list:
+   - If it starts with ">=", then the comparator is ">=".
+   - If it starts with "<",  then the comparator is "<".
 
-    - Identify the <version-constraint> comparator and version based on the
-      start of the <version-constraint> in this sequence:
+   - Remove the comparator from <version-constraint> string, and the
+     remaining string is the version.
 
-       - If it starts with "=",  then the comparator is "="
-       - If it starts with "!=", then the comparator is "!=".
-       - If it starts with "<=", then the comparator is "<=".
-       - If it starts with ">=", then the comparator is ">=".
-       - If it starts with "<",  then the comparator is "<".
-       - If it starts with ">",  then the comparator is ">".
-       - Else the comparator is "=" (default) and the
-         version is the full <version-constraint> string.
+  - Otherwise the version is the full <version-constraint> string (which implies
+    and equality comparator of "=")
 
-    - After the operation and removing the comparator from <version-constraint>
-      string, the remaining string is the version.
+  - A tool should validate and report an error if the version is empty.
 
-    - Validate that the version is not empty.
+  - If the version contains a percent "%" character, apply URL quoting rules
+    to unquote this string.
 
-    - If the version contains a percent "%" character, apply URL quoting rules
-      to unquote this string.
+  - Append the parsed (comparator, version) to the constraints list.
 
-    - Append the comparator and version of this constraint to the inner list
-      of constraints.
+Finally:
 
-  - Append the accumulated list of (comparator and version) that must apply to
-    the top level list of constraints.
-
-- Finally return the <versioning-scheme> and the nested list of <version-constraint>
+- deduplicate and sort the list of (comparator, version) using the canonical order.
+- return the <versioning-scheme> and this list of constraints.
```

Comment:
Beyond parsing, should we include some simple pseudocode demonstrating how to evaluate these without any ambiguity?

Reply:
Yes, this would be awesome.

Reply:
This was added in the latest push together with pseudocode for validation and deduplication.
Thanks!